src/client_side_reply.cc

```diff
@@ -1486,15 +1487,14 @@ clientReplyContext::identifyStoreObject()
 
     // client sent CC:no-cache or some other condition has been
     // encountered which prevents delivering a public/cached object.
-    // XXX: The above text does not match the condition below. It might describe
-    // the opposite condition, but the condition itself should be adjusted
-    // (e.g., to honor flags.noCache in cache manager requests).
-    if (!r->flags.noCache || r->flags.internal) {
+    if (r->flags.noCache) {
+        // no-cache requests skip Store lookups
+        identifyFoundObject(nullptr, "no-cache");
+    } else {
+        // cache manager requests ought to have flags.noCache (handled above)
+        Assure(!ForSomeCacheManager(r->url.path()));
         const auto e = storeGetPublicByRequest(r);
```

Comment:
Clarify the scope of the above comment/Assure combo by separating that code from the primary handling code:

```suggestion

        const auto e = storeGetPublicByRequest(r);
```

Reply:
I changed the condition in 6ae2592e, so this looks irrelevant now.